src/tlv/TlvNumber.ts

```diff
@@ -78,6 +77,12 @@ export class TlvNumberSchema extends TlvNumericSchema<number> {
             minValue(max, this.max),
         );
     }
+
+    override validate(value: number): void {
+        if (typeof value !== "number") throw new Error(`Expected number, got ${typeof value}.`);
+        if (this.min !== undefined && value < this.min) throw new Error(`Invalid value: ${value} is below the minimum, ${this.min}.`);
```

Comment:
Extract this into a validateBoundary method part of TlvNumericSchema to avoid repeating this code.

Reply:
makes sense